anacron/runjob.c

```diff
@@ -247,13 +254,38 @@ launch_mailer(job_rec *jr)
 	/* fdflags = fcntl(0, F_GETFL); fdflags &= ~O_APPEND; */
 	/* fcntl(0, F_SETFL, fdflags ); */
 
-	/* Here, I basically mirrored the way /usr/sbin/sendmail is called
-	 * by cron on a Debian system, except for the "-oem" and "-or0s"
-	 * options, which don't seem to be appropriate here.
-	 * Hopefully, this will keep all the MTAs happy. */
-	execl(SENDMAIL, SENDMAIL, "-FAnacron", "-odi",
-	      jr->mailto, (char *)NULL);
-	die_e("Can't exec " SENDMAIL);
+        /* Get SENDMAIL variable from the environment if set. 
+        /* If not, will use the predefined SENDMAIL from global.h */
+        mailer_full = getenv("SENDMAIL");
+        
+        if (mailer_full == NULL) {
+            /* Here, I basically mirrored the way /usr/sbin/sendmail is called
+            * by cron on a Debian system, except for the "-oem" and "-or0s"
+            * options, which don't seem to be appropriate here.
+            * Hopefully, this will keep all the MTAs happy. */
+            execl(SENDMAIL, SENDMAIL, "-FAnacron", "-odi",
+                jr->mailto, (char *)NULL);
+            die_e("Can't exec " SENDMAIL); 
+        } else {
+            
+            /* Splitting into binary path and command line parameters*/
+            
+            complain("mailer_full - `%s`", mailer_full);
+           
+            strcat(mailer_full," ");
+            r = match_rx("(^[^ ]*) (.*)", mailer_full, 2, &mailer, &mailer_args);
```

Comment:
Unfortunately this is more complicated as for executing the mailer via execl the arguments would have to be split. Or perhaps better would be to just allow a few mailer arguments in separate environment variables. `SENDMAILARG1` `SENDMAILARG2` `SENDMAILARG3`....

Reply:
> Unfortunately this is more complicated as for executing the mailer via execl the arguments would have to be split. Or perhaps better would be to just allow a few mailer arguments in separate environment variables. `SENDMAILARG1` `SENDMAILARG2` `SENDMAILARG3`....

Is that... even acceptable to shift the task of separating arguments to the user? I mean... that's kinda rude even for me. Is that okay?

Reply:
It is a problem to properly separate the arguments in C because you'd basically have to re-implement the shell argument separator parser to make it really useful. I.e. you'd have to support at least `"`, `'` as quotes and then you also need to somehow support escaping these quote characters, etc.